### PR TITLE
chore: upgrade OT core to v0.54.0

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: pkg/receiver/rawk8seventsreceiver/receiver_test.go [hunk 1]
```diff
@@ -408,6 +408,8 @@ func TestStorage(t *testing.T) {
 	host := storagetest.NewStorageHost(t, storageDir, "test")
 	assert.NoError(t, receiver.Start(ctx, host))
 
+	time.Sleep(10 * time.Millisecond)
+
```
Comment (sumo-drosiek) [inline pkg/receiver/rawk8seventsreceiver/receiver_test.go:412]: :fearful:

File: README.md [hunk 2]
```diff
@@ -78,7 +78,6 @@ The rest of the components in the table are pure upstream OpenTelemetry componen
 |                    [otlp][otlpreceiver]                    |                                                              |                                        |                                              |
 |               [podman_stats][podmanreceiver]               |                                                              |                                        |                                              |
 |              [postgresql][postgresqlreceiver]              |                                                              |                                        |                                              |
-|         [prometheus_exec][prometheusexecreceiver]          |                                                              |                                        |                                              |
```
Comment (sumo-drosiek) [inline README.md:81]: It was deprecated, not removed yet:

https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/prometheusexecreceiver

Comment (astencel-sumo) [reply README.md:81]: It's removed from the contrib distribution. https://github.com/open-telemetry/opentelemetry-collector-releases/pull/152

My understanding is we include all the components that are included in otelcol and otelcol-contrib distributions, not necessarily all the components that are included in the core or contrib source repository.

File: pkg/receiver/rawk8seventsreceiver/receiver_test.go [hunk 3]
```diff
@@ -408,6 +408,8 @@ func TestStorage(t *testing.T) {
 	host := storagetest.NewStorageHost(t, storageDir, "test")
 	assert.NoError(t, receiver.Start(ctx, host))
 
+	time.Sleep(10 * time.Millisecond)
```
Comment (pdelewski) [inline pkg/receiver/rawk8seventsreceiver/receiver_test.go:411]: What's the reason behind this?

File: pkg/receiver/rawk8seventsreceiver/receiver_test.go [hunk 4]
```diff
@@ -343,7 +343,7 @@ func TestNoStorage(t *testing.T) {
 	// Both events should be picked up by the receiver.
 	assert.Eventually(t, func() bool {
 		return assert.Equal(t, 2, logsSink.LogRecordCount())
-	}, time.Second, 100*time.Millisecond)
+	}, 100*time.Millisecond, 10*time.Millisecond)
```
Comment (pdelewski) [inline pkg/receiver/rawk8seventsreceiver/receiver_test.go:346]: Why do we needed to use `Millisecond` here?

**Review by pdelewski — APPROVED** [review]

File: pkg/receiver/rawk8seventsreceiver/receiver_test.go (hunk 3 above)
Comment (astencel-sumo) [reply pkg/receiver/rawk8seventsreceiver/receiver_test.go:411]: This is to ensure that the second event actually comes in _after_ the first event.

**Pull request merged**